File: backend/app/models/bot.py
import uuid
from datetime import datetime

from sqlalchemy import String, Text, Boolean, DateTime, ForeignKey, JSON
from sqlalchemy.dialects.postgresql import UUID
from sqlalchemy.orm import Mapped, mapped_column, relationship

from app.database import Base


class Bot(Base):
    __tablename__ = "bots"

    id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), primary_key=True, default=uuid.uuid4)
    creator_id: Mapped[uuid.UUID] = mapped_column(UUID(as_uuid=True), ForeignKey("users.id"))
    name: Mapped[str] = mapped_column(String(255))
    avatar_url: Mapped[str | None] = mapped_column(String(1024), nullable=True)
    system_prompt: Mapped[str] = mapped_column(Text, default="You are a helpful AI assistant.")
    voice_name: Mapped[str] = mapped_column(String(50), default="Kore")
    integrations_config: Mapped[dict | None] = mapped_column(JSON, nullable=True, default=dict)
    is_default: Mapped[bool] = mapped_column(Boolean, default=False)
    created_at: Mapped[datetime] = mapped_column(DateTime, default=datetime.utcnow)

    creator = relationship("User", back_populates="bots")
    chats = relationship("Chat", back_populates="bot", lazy="selectin")

File: app/lib/models/bot.dart
class Bot {
  final String id;
  final String creatorId;
  final String name;
  final String? avatarUrl;
  final String systemPrompt;
  final String voiceName;
  final Map<String, dynamic>? integrationsConfig;
  final int? proactiveMinutes;
  final bool isDefault;
  final DateTime createdAt;

  Bot({
    required this.id,
    required this.creatorId,
    required this.name,
    this.avatarUrl,
    required this.systemPrompt,
    this.voiceName = 'Kore',
    this.integrationsConfig,
    this.proactiveMinutes = 0,
    required this.isDefault,
    required this.createdAt,
  });

  factory Bot.fromJson(Map<String, dynamic> json) => Bot(
        id: json['id'],
        creatorId: json['creator_id'],
        name: json['name'],
        avatarUrl: json['avatar_url'],
        systemPrompt: json['system_prompt'],
        voiceName: json['voice_name'] ?? 'Kore',
        integrationsConfig: json['integrations_config'],
        proactiveMinutes: json['proactive_minutes'] == null
            ? null
            : (json['proactive_minutes'] is int
                ? json['proactive_minutes']
                : int.tryParse('${json['proactive_minutes']}')),
        isDefault: json['is_default'] ?? false,
        createdAt: DateTime.parse(json['created_at']),
      );
}
